### PR TITLE
SDK enhances communication, app supports event monitoring

### PR DESCRIPTION
**Pull request by zjy365**
### Summary
:sparkles::wrench::memo:

This pull request adds new features and APIs to the `client-sdk` project and integrates it with the `desktop` and `applaunchpad` projects. It updates the dependencies, configuration, and documentation of the `client-sdk` project and modifies some components and pages of the other projects to enable communication and authentication. It also improves the formatting and readability of the `README_zh.md` file for the `client-sdk` project.

> _We are the masters of the SDK_
> _We unleash the power of the API_
> _We integrate and communicate across the sealos_
> _We are the client of doom and chaos_

### Walkthrough
*  Update the version and add the dependency of `sealos-desktop-sdk` in `frontend/client-sdk/package.json` ([link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-5c4adf804d6737cb0d8bcdeece17aae8d788b74d30e75af24dcad0e44a6e374fL3-R3), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-5c4adf804d6737cb0d8bcdeece17aae8d788b74d30e75af24dcad0e44a6e374fL31-R33))
* Implement new features and APIs for the client-sdk project by adding and modifying imports, types, properties, methods, and enums in `app.ts`, `api.ts`, `event.ts`, `master.ts`, and `cookieUtils.ts` ([link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-ee3a4759a8b771f1c4c5e357e12615948310ea87014cdebc85a8e666e837d3c6L3-R7), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-ee3a4759a8b771f1c4c5e357e12615948310ea87014cdebc85a8e666e837d3c6L14-R20), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-ee3a4759a8b771f1c4c5e357e12615948310ea87014cdebc85a8e666e837d3c6L65-R86), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-ee3a4759a8b771f1c4c5e357e12615948310ea87014cdebc85a8e666e837d3c6R107-R110), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-ee3a4759a8b771f1c4c5e357e12615948310ea87014cdebc85a8e666e837d3c6R120-R159), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-9f2b7c21e5b0abb034a7aec5b52c7fab8b38f83ed951f05a99a28dc24dab3215L3-R4), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-a39388e0ed3f5cf4faf0a5cfacdc6d58fa501caa0fded7a987996954dbe54d49L2-R3), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-a6c87f8f20ac55bd024ec33bffb6c4c5358b48e6b3083a702281ccb19de0212dL1-R9), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-a6c87f8f20ac55bd024ec33bffb6c4c5358b48e6b3083a702281ccb19de0212dL11-R18), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-a6c87f8f20ac55bd024ec33bffb6c4c5358b48e6b3083a702281ccb19de0212dR173-R209), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-f7026d92d54b0c0f5140b260dd794babff8784d0ac771e26b687c69715f7ad82R1-R15))
* Integrate the client-sdk project with the desktop and applaunchpad projects by adding and modifying imports, calls, variables, and logic in `index.tsx` and `_app.tsx` ([link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-fd7e2f708a39a8f0750cd167c304371e36be9ffec2c36066025620dd45b1a1c8R4-R5), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-fd7e2f708a39a8f0750cd167c304371e36be9ffec2c36066025620dd45b1a1c8R50-R56), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-ebd69d41e82f5d5fbc4c8884ac0789e5e3e743e88c10031cebbf19d53120ea02R10), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-ebd69d41e82f5d5fbc4c8884ac0789e5e3e743e88c10031cebbf19d53120ea02L16-R21), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-ebd69d41e82f5d5fbc4c8884ac0789e5e3e743e88c10031cebbf19d53120ea02R41), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-ebd69d41e82f5d5fbc4c8884ac0789e5e3e743e88c10031cebbf19d53120ea02L90-R111))
* Improve the local debugging workflow for the client-sdk project by adding the installation of nodemon as a global dependency in `README_zh.md` ([link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-0004ff691350cd72e16dd83e496c9a8ebf92b88441e3c4afbc678958ef94bcaaL6-R13))
* Improve the formatting and readability of the documentation by adding and removing empty lines and spaces in `README_zh.md` ([link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-0004ff691350cd72e16dd83e496c9a8ebf92b88441e3c4afbc678958ef94bcaaR19), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-0004ff691350cd72e16dd83e496c9a8ebf92b88441e3c4afbc678958ef94bcaaL39-R46), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-0004ff691350cd72e16dd83e496c9a8ebf92b88441e3c4afbc678958ef94bcaaL61-R66), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-0004ff691350cd72e16dd83e496c9a8ebf92b88441e3c4afbc678958ef94bcaaL75-R88), [link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-0004ff691350cd72e16dd83e496c9a8ebf92b88441e3c4afbc678958ef94bcaaL87-R121))
* Improve the dependency management and configuration of the applaunchpad project by updating the lockfileVersion and adding some settings in `pnpm-lock.yaml` ([link](https://github.com/labring/sealos/pull/3311/files?diff=unified&w=0#diff-d36a647289ca0f89be547a4f8cc117ce2a99a9b9815375ba25dfa84ea02bb390L1-R6))

### CODE REVIEW

File: frontend/client-sdk/src/master.ts [hunk 1]
```diff
@@ -163,6 +170,43 @@ class MasterSDK {
...
+        source,
+        origin,
+        messageId: data.messageId,
+        success: true,
+        data: {
+          lng: lng
+        }
+      });
+    } else {
+      this.replyAppMessage({
```
Comment (c121914yu) [inline frontend/client-sdk/src/master.ts:190]: default reply en

File: frontend/providers/applaunchpad/src/pages/_app.tsx [hunk 2]
```diff
@@ -87,7 +90,24 @@ const App = ({ Component, pageProps }: AppProps) => {
     return () => {
       setLastRoute(router.asPath);
     };
-  }, [router.pathname, setLastRoute]);
+  }, [router.asPath, router.pathname, setLastRoute]);
```
Comment (c121914yu) [inline frontend/providers/applaunchpad/src/pages/_app.tsx:93]: Do not add asPath

File: frontend/providers/applaunchpad/src/pages/_app.tsx [hunk 3]
```diff
@@ -87,7 +90,24 @@ const App = ({ Component, pageProps }: AppProps) => {
...
+
+  const changeI18n = (data: any) => {
```
Comment (c121914yu) [inline frontend/providers/applaunchpad/src/pages/_app.tsx:95]: Method is merged directly into useEffect.

File: frontend/client-sdk/README_zh.md [hunk 4]
```diff
@@ -72,41 +77,45 @@ useEffect(() => {
...
 <br/>
 
 ### Desktop 和 APP 之间的通信说明
-Desktop 容器通过 Iframe 挂载APP，两者之间通过 PostMessage 进行单向通信。SDK分为 Master SDK 和 APP SDK。
 
-Master SDK 放到了 Desktop 项目中运行，APP 开发者不需要关系 Master SDK 的内容。  
+Desktop 容器通过 Iframe 挂载 APP，两者之间通过 PostMessage 进行单向通信。SDK 分为 Master SDK 和 APP SDK。
 
-APP SDK 中，封装了和 Master SDK 通信的方法，可以直接通过 await ***() 的形式，同步的书写对应业务代码。  
+Master SDK 放到了 Desktop 项目中运行，APP 开发者不需要关系 Master SDK 的内容。
```
Comment (zzjin) [inline frontend/client-sdk/README_zh.md:98]: wrong spelling

**Review by zzjin — APPROVED** [review]

File: frontend/client-sdk/src/app.ts [hunk 5]
```diff
@@ -100,6 +118,46 @@ class ClientSDK {
...
+  removeAppEventListen(name: string) {
+    this.eventBus.delete(name);
+  }
+
+  /**
+   * run app event bus
+   */
+  private runAppEvents(data: AppMessageType) {
+    if ('eventName' in data) {
+      if (!this.eventBus.has(data.eventName)) {
```
Comment (c121914yu) [inline frontend/client-sdk/src/app.ts:151]: if ('eventName' in data && this.eventBus.has(data.eventName) )

**Pull request merged**